src/components/postgresConnect.ts

```diff
@@ -316,6 +316,82 @@ class PostgresDB {
       return false
     }
   }
+
+  public getTemplateStages = async (templateId: number) => {
+    const text = 'SELECT id, number, title FROM template_stage WHERE template_id = $1'
+    try {
+      const result = await this.query({ text, values: [templateId] })
+      return result.rows
+    } catch (err) {
+      console.log(err.message)
+      throw err
+    }
+  }
+
+  public getCurrentStageHistory = async (applicationId: number) => {
+    const text =
+      'SELECT id, stage_id FROM application_stage_history WHERE application_id = $1 and is_current = true'
+    try {
+      const result = await this.query({ text, values: [applicationId] })
+      return result.rows[0]
+    } catch (err) {
+      console.log(err.message)
+      throw err
+    }
+  }
+
+  public addNewStageHistory = async (applicationId: number, stageId: number) => {
+    // Note: switching is_current of previous stage_histories to False is done automatically by a Postgres trigger function
+    const text =
+      'INSERT into application_stage_history (application_id, stage_id, time_created) VALUES ($1, $2, CURRENT_TIMESTAMP) RETURNING id'
+    try {
+      const result = await this.query({ text, values: [applicationId, stageId] })
+      return result.rows[0].id
+    } catch (err) {
+      console.log(err.message)
+      throw err
+    }
+  }
+
+  public getCurrentStatusHistory = async (applicationId: number) => {
+    const text = `SELECT application_status_history.id,
+      application_stage_history_id, status FROM
+      application_status_history JOIN application_stage_history ON
+      application_stage_history_id = application_stage_history.id WHERE
+      application_id = $1 AND application_status_history.is_current = true;`
```

Comment:
> This is linked to what I commented on `getCurrentStageHistory` method. Why not using the view `application_stage_status` instead?

Because you can't WRITE to a VIEW table, so the query to update the status_history `is_current` is still pretty nasty even when referring to the `application_stage_status` View.  I've used @andreievg 's auto-generated `application_id` column, and that works a treat :) 

However, I will use the VIEW table for the `get...` query below, so thanks for that suggestion.